src/test/java/httpServer/ListenerAndResponderTest.java

```diff
@@ -38,9 +38,10 @@ void setUp() throws IOException {
         mockServerSocket = new MockServerSocketWrapper(socket);
         mockPrintWriter =  new MockPrintWriterWrapper();
         mockBufferedReader = new MockBufferedReaderWrapper(input);
+        RequestParser requestParser = new RequestParser();
```

Comment:
You probably want to mock this out and replace `testReadClientInputCallsMockBufferedReader()` with `testReadClientInputCallsRequestParser()`.

Reply:
I made a mock to make sure RequestParser was called.